### PR TITLE
ci(github): add provenance for images

### PR DESCRIPTION
**Pull request by lahabana**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .github/workflows/_provenance.yaml [hunk 1]
```diff
@@ -20,3 +28,25 @@ jobs:
...
+  # ENV option is [not supported](https://github.com/orgs/community/discussions/26671)
+  # Reusable workflow doesn't support exrernal COSIGN_REPOSITORY via input / env variable
+  # TODO:
+  #   Split provenance jobs for internal / official releases when repositories are split
+  images-provenance:
+    strategy:
+      fail-fast: true
+      matrix:
+        IMAGE: ${{ fromJSON(inputs.images) }}
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@07e64b653f10a80b6510f4568f685f8b7b9ea830 #v1.9.0
```
Comment (saisatishkarra) [inline .github/workflows/_provenance.yaml:45]: This needs to be a tag and use v1.10.0 for cosign to work without `invalid key` error against sig store TUF verification

**Review by slonka — COMMENTED** [review]
Did you run it manually / can you link to a successful run?

**Review by jakubdyszkiewicz — APPROVED** [review]

File: .github/workflows/_build_publish.yaml [hunk 2]
```diff
@@ -76,6 +76,7 @@ jobs:
             ./build/distributions/out/*.tar.gz
             ./build/distributions/out/*.sha256
             !./build/distributions/out/*.tar.gz.sha256
+          retention-days: ${{ github.event_name == 'pull_request' && 1 || 30 }}
```
[the remark this answers is not in the document]
Comment (lahabana) [reply .github/workflows/_build_publish.yaml:79]: We don't really use them tbh. I've just done the same as we do for helm releases

**Comment by lahabana** [comment]
It ran here. But the provenance is tricky as it doesn't really work on non release stuff. I think a follow up should enable us to make it work.

**Pull request merged**